### PR TITLE
Issue 890 - close Jetty related issues, update docker-compose.yml

### LINKED ISSUE #407: Upgrade the version of Jetty to >= 9.4.8.v20171121
We need to upgrade to the latest version of Jetty.

Make sure that:
* The project still builds
* The tests are still passing

The version is controlled via the `strongbox-parent/pom.xml`.

The webapp can be executed using:

```
mvn clean install
cd strongbox-web-core
mvn clean package -Pjetty-block
```

If Jetty is starting the webapp correctly, then things should be alright. To test things, try accessing:
http://localhost:48080/configuration/strongbox/xml with `admin`/`password` -- an XML with the configuration of the server should be presented, upon successful login.

### LINKED ISSUE #422: Update jetty to version >= 9.4.7.v20170914
We need to upgrade to the latest version of Jetty.

Make sure that:

* The project still builds
* The tests are still passing
* The version is controlled via the strongbox-parent/pom.xml.

The webapp can be executed using:

```
mvn clean install
cd strongbox-web-core
mvn clean package -Pjetty-block
```

If Jetty is starting the webapp correctly, then things should be alright. To test things, try accessing:
http://localhost:48080/configuration/strongbox/xml with admin/password -- an XML with the configuration of the server should be presented, upon successful login.

Note : Jetty 9.4.7 and 9.4.8 builds the web app correctly. The tests pass too. The issue is that the Jetty server does not start at the end for deploying web app. Note that jetty v9.4.8 requires maven 3.5.0.

### LINKED ISSUE #890: Upgrade Jetty to 9.4.12.v20180830
Using the current Jetty version of 9.4.6 there is a RuntimeException when starting Jetty, see here https://gist.github.com/bamarch/72bba0d8ca988902b2d7e9fd9bcdd6c4

The issue is related to compatibility with JDK 9 features that are present in some dependency jars.

The solution is to either downgrade the jars or upgrade Jetty (to a version >= 9.4.9). 

Relates to:
https://github.com/strongbox/strongbox/issues/407 
https://github.com/strongbox/strongbox/issues/421 
https://github.com/strongbox/strongbox/issues/422
https://github.com/strongbox/strongbox/pull/804

### PR DESCRIPTION
**Pull request by bamarch**
Jetty upgrade not needed: the project was migrated to spring boot.

The docker-compose file needed updating to reflect this.

There are also a number of files generated when running `docker-compose up` and I added those paths to .gitignore

Issues / PR related to Jetty:
closes #890 
closes #407
closes #422
closes #804

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,6 +39,10 @@ tmp/
 logs/
 *.log
 
+# generated
+strongbox-web-core/strongbox-vault
+strongbox-web-core/strongbox
```
Comment (carlspring) [inline .gitignore]: Could you please revert these two lines? Thanks for pointing this out. We'll fix it. It shouldn't be ignored through here.

Comment (bamarch) [reply .gitignore]: ok done

File: docker-compose.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ services:
          mkdir -p /workspace/.m2/repositories
          mvn clean install -DskipTests -Dmaven.repo.local=/workspace/.m2/repositories
          cd strongbox-web-core
-         mvn clean package -Pjetty-block -Dmaven.repo.local=/workspace/.m2/repositories
+         mvn spring-boot:run -Dmaven.repo.local=/workspace/.m2/repositories
```
[the remark this answers is not in the document]
Comment (steve-todorov) [reply docker-compose.yml]: Thanks for fixing this :+1:  :)

**Comment by bamarch** [comment]
I signed the ICLA and sent it by email, so I also updated contributing.md here

**Comment by carlspring** [comment]
Thanks for your fixes and welcome to the project! :)

**Pull request merged**